sdk/keyvault/keyvault-certificates/src/index.ts

```diff
@@ -842,19 +918,20 @@ export class CertificateClient {
     certificateName: string,
     certificatePolicy: CertificatePolicy,
     options: CreateCertificateOptions = {}
-  ): Promise<Certificate> {
-    const span = this.createSpan("createCertificate", options);
+  ): Promise<KeyVaultCertificate> {
```

Comment:
This method doesn't seem to fully plumb all the ROB fields. Consider what would happen if we added entries to ROB that core-http will handle. Those will need to pass through appropriately.

Reply:
I have updated this method, but I'm not sure if I've addressed this feedback. Please let me know what I'm missing! I'm not sure if I can parse what ROB means.

Reply:
by ROB he means RequestOptionsBase

Reply:
It should be addressed now! Either that or the other clients are missing something too. This specific code is now on par with keys and secrets.

Reply:
@bterlson The code that existed before @sadasant made the change to address your comment did plumb through all of the ROB fields. The `...this.setParentSpan()` takes care of that.

@sadasant The change you made to make things in par with keys & secrets doesnt exactly cater to the certificates. 

For example, `CreateKeyOptions` contained `enabled`, `notBefore` and `expiresOn` properties, therefore extracting them out, setting them to `keyAttributes` before calling the internal `createKey` made sense.

But, `CreateCertificateOptions` has only `enabled` and not `expiresOn` or `notBefore` on it.
`expiresOn` and `notBefore` are defined on the `certificateAttributes` property.

So, your intention to mimic keys, ends up setting the `expires` and `notBefore` properties as always undefined even if user provided them via the `certificateAttributes` property.

I believe that the code that existed before [this commit](https://github.com/Azure/azure-sdk-for-js/pull/5991/commits/29fb97142410f474410207501802a05a87a107e0) (except the `requestOptions.requestrequestOptions` being just `requestOptions` was the right one to use.

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/5991/commits/32ac2e97b8ed98d87f315cd20a9db30c2e22e49d resolves my concerns.